lib/archethic/replication.ex

```diff
@@ -106,7 +106,19 @@ defmodule Archethic.Replication do
           tx
           |> stream_previous_chain(download_nodes)
           |> Stream.reject(&Enum.empty?/1)
-          |> Stream.each(&TransactionChain.write/1)
+          |> Stream.flat_map(& &1)
+          |> Stream.each(fn tx = %Transaction{address: address} ->
+            TransactionChain.write_transaction(tx)
+
+            # There is some case where a transaction is not replicated while it should
+            # because of some latency or network issue. So when we replicate a past chain
+            # we also ingest the transaction if we are storage node of it
+
+            if address
```

Comment:
I think you can extract into a variable to make it easier to read
Long instructions in a condition statement are harder to understand.